### PR TITLE
Adds new optional arguments to `DataFrame.(read/write)_ipc`

### LINKED ISSUE #147: Document use cases of Arrow IPC format
So if I get this right, is this mostly a way to serialize/deserialize dataframes? What are the use cases? Maybe we should have one or two paragraphs in the docs breaking this a bit down? :)

_Originally posted by @josevalim in https://github.com/elixir-nx/explorer/issues/144#issuecomment-1084166397_

### PR DESCRIPTION
**Pull request by kimjoaoun**
This pull request adds features related to the reading and writing of Apache Arrow's IPC files. Apache IPC is a language-agnostic columnar data structure that can be used to store information which can be read between several programming languages ([list of supported languages](https://arrow.apache.org/docs/status.html)). 

- [x] Better documentation for IPC (closes #147)
- [x] Adds optional arguments for `DataFrame.read_ipc()`
- [x] Adds optional arguments for `DataFrame.write_ipc()`

### CODE REVIEW

File: lib/explorer/data_frame.ex [hunk 1]
```diff
@@ -129,17 +129,65 @@ defmodule Explorer.DataFrame do
 
   @doc """
   Reads a IPC file into a dataframe.
+
+    ## Options
+
+      * `with_columns` - List with name of columns to be selected.
```
Comment (josevalim) [inline lib/explorer/data_frame.ex:135]: Is `with_columns` the name we typically use for picking columns in Explorer?

Comment (kimjoaoun) [reply lib/explorer/data_frame.ex:135]: We normally use "`select`", do you think "`with_selection`" or simply "`select`" is a better choice?

File: lib/explorer/data_frame.ex [hunk 2]
```diff
@@ -129,17 +129,65 @@ defmodule Explorer.DataFrame do
...
+  def read_ipc!(filename, opts \\ []) do
+    case read_ipc(filename, opts) do
+      {:ok, df} -> df
+      {:error, error} -> raise "#{error}"
+    end
+  end
 
   @doc """
   Writes a dataframe to a IPC file.
+  Apache IPC is a language-agnostic columnar data structure that can be used to store data frames.
```
Comment (philss) [inline lib/explorer/data_frame.ex:168]: I would only add a line here. It's common to separate the ["title"](https://hexdocs.pm/elixir/writing-documentation.html#recommendations) from the body of the doc.

```suggestion

  Apache IPC is a language-agnostic columnar data structure that can be used to store data frames.
```

File: lib/explorer/data_frame.ex [hunk 3]
```diff
@@ -129,17 +129,65 @@ defmodule Explorer.DataFrame do
 
   @doc """
   Reads a IPC file into a dataframe.
+
+    ## Options
+
+      * `columns` - List with name of columns to be selected. Defaults to all columns.
+      * `projection` - List with the index of columns to be selected.
```
Comment (philss) [inline lib/explorer/data_frame.ex:136]: Does it defaults to all columns as well?

**Review by philss — APPROVED** [review]
LGTM :shipit: 

Only two minor comments

File: lib/explorer/data_frame.ex (hunk 3 above)
Comment (kimjoaoun) [reply lib/explorer/data_frame.ex:136]: Yes, it does. I'll add this information.

File: lib/explorer/data_frame.ex [hunk 4]
```diff
@@ -129,17 +129,66 @@ defmodule Explorer.DataFrame do
 
   @doc """
   Reads a IPC file into a dataframe.
+
+    ## Options
+
+      * `columns` - List with name of columns to be selected. Defaults to all columns.
+      * `projection` - List with the index of columns to be selected. Defaults to all columns.
```
Comment (josevalim) [inline lib/explorer/data_frame.ex:136]:
```suggestion
  ## Options

    * `columns` - List with name of columns to be selected. Defaults to all columns.
    * `projection` - List with the index of columns to be selected. Defaults to all columns.
```

File: lib/explorer/data_frame.ex [hunk 5]
```diff
@@ -129,17 +129,66 @@ defmodule Explorer.DataFrame do
...
   @doc """
   Writes a dataframe to a IPC file.
+
+  Apache IPC is a language-agnostic columnar data structure that can be used to store data frames.
+  It excels as a format for quickly exchange data between different programming languages.
+
+  ## Options
+
+  * `compression` - Sets the algorithm used to compress the IPC file.
+  It accepts `"ZSTD"` or `"LZ4"` compression. (default: `nil`)
```
Comment (josevalim) [inline lib/explorer/data_frame.ex:175]:
```suggestion
    * `compression` - Sets the algorithm used to compress the IPC file.
      It accepts `"ZSTD"` or `"LZ4"` compression. (default: `nil`)
```

File: lib/explorer/data_frame.ex (hunk 1 above)
Comment (cigrainger) [reply lib/explorer/data_frame.ex]: I agree with @josevalim :+1:

Comment (kimjoaoun) [reply lib/explorer/data_frame.ex]: Nice!

Made those changes on https://github.com/elixir-nx/explorer/commit/26d05270590af6ef8ba500616a254ef330427acb and https://github.com/elixir-nx/explorer/pull/152/commits/e62e61cf1654fa0eff9e31be479d7bf96964707f

**Review by cigrainger — APPROVED** [review]
Looks good to me! Thanks for this @kimjoaoun

**Pull request merged**